azurerm/internal/services/datamigration/data_source_data_migration_service.go

```diff
@@ -0,0 +1,88 @@
+package datamigration
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmDataMigrationService() *schema.Resource {
```

Comment:
I think we can remove this Data Source, since this isn't something users are likely to use directly from Terraform?

Reply:
Acutally there are some upcoming resources on top of DMS (see #5257 for more info). It might be some rare cases that user A created a DMS service and define different DMS projects for different migration scenario into different terraform modules, inside where user might just want to pass in an input variable indicating in which DMS server to create the DMS project/tasks.

Reply:
So I think we can just leave the data source there and remove the `kind` properties for now?